Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/transform/audit/COREX12RealtimeTransforms.java

```diff
@@ -0,0 +1,342 @@
+        auditMsg.setEventIdentification(oEventIdentificationType);
+
+        AuditMessageType.ActiveParticipant participant = getActiveParticipant(assertion.getUserInfo());
+        auditMsg.getActiveParticipant().add(participant);
+
+        /* Assign ParticipationObjectIdentification */
+        ParticipantObjectIdentificationType participantObject = getParticipantObjectIdentificationType(response.getPayloadID());
+        auditMsg.getParticipantObjectIdentification().add(participantObject);
+
+        /* Create the AuditSourceIdentifierType object */
+        String communityId = oXDR.getMessageCommunityIdFromRequest(assertion, target, direction, _interface);
+        AuditSourceIdentificationType auditSource = getAuditSourceIdentificationType(communityId);
+        auditMsg.getAuditSourceIdentification().add(auditSource);
+
+        result.setAuditMessage(auditMsg);
+        result.setDirection(direction);
+        result.setInterface(_interface);
+
+        LOG.debug("end transformResponseToAuditMsg() -- NHIN");
+
+        return result;
+    }
+
+    private boolean areRequiredCOREX12RealtimeFiledsNull(COREEnvelopeRealTimeResponse msg, AssertionType assertion) {
+        try {
+
+            if (assertion == null) {
+                LOG.error("Assertion object is null");
+                return true;
+            }
+            if (msg == null) {
+                LOG.error("COREEnvelopeRealTimeRequest object is null");
+                return true;
+            }
+
+            if (oXDR.areRequiredUserTypeFieldsNull(assertion)) {
+                LOG.error("One of more UserInfo fields from the Assertion object were null.");
+                return true;
+            }
+
+            if (areRequiredRequestFieldsNull(msg)) {
+                LOG.error("One or more fields in CORE X12 Realtime Request object were null.");
+                return true;
+            }
+
+
+        } catch (Exception exp) {
+            LOG.error("Encountered Error: " + exp.getMessage());
+            return true;
+        }
+        return false;
+    }
+
+    private boolean areRequiredCOREX12RealtimeFiledsNull(COREEnvelopeRealTimeRequest msg, AssertionType assertion) {
+        try {
+
+            if (assertion == null) {
+                LOG.error("Assertion object is null");
+                return true;
+            }
+            if (msg == null) {
+                LOG.error("COREEnvelopeRealTimeRequest object is null");
+                return true;
+            }
+
+            if (oXDR.areRequiredUserTypeFieldsNull(assertion)) {
+                LOG.error("One of more UserInfo fields from the Assertion object were null.");
+                return true;
+            }
+
+            if (areRequiredRequestFieldsNull(msg)) {
+                LOG.error("One or more fields in CORE X12 Realtime Request object were null.");
+                return true;
+            }
+
+        } catch (Exception exp) {
```

Comment:
Updated. No exception, this is following XDR may be need to update XDR project as well sometime. Creating a ticket.